src/main/java/modbus/IpConnection.java

```diff
@@ -0,0 +1,200 @@
+package modbus;
+
+import org.dsa.iot.dslink.node.Node;
+import org.dsa.iot.dslink.node.Permission;
+import org.dsa.iot.dslink.node.actions.Action;
+import org.dsa.iot.dslink.node.actions.ActionResult;
+import org.dsa.iot.dslink.node.actions.Parameter;
+import org.dsa.iot.dslink.node.value.Value;
+import org.dsa.iot.dslink.node.value.ValueType;
+import org.dsa.iot.dslink.util.handler.Handler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.serotonin.modbus4j.ModbusFactory;
+import com.serotonin.modbus4j.ModbusMaster;
+import com.serotonin.modbus4j.exception.ModbusInitException;
+import com.serotonin.modbus4j.ip.IpParameters;
+
+public class IpConnection extends ModbusConnection {
+
+	private static final Logger LOGGER;
+
+	static {
+		LOGGER = LoggerFactory.getLogger(IpConnection.class);
+	}
+
+	static final String ATTR_NAME = "name";
+	static final String ATTR_TRANSPORT_TYPE = "transport type";
+	static final String ATTR_HOST = "host";
+	static final String ATTR_PORT = "port";
+	static final String ADD_IP_DEVICE_ACTION = "add ip device";
+
+	IpConnection(ModbusLink link, Node node) {
+		super(link, node);
+	}
+
+	ModbusMaster getMaster() {
+		if (this.master != null) {
+			return this.master;
+		}
+
+		if (reconnectFuture != null) {
+			reconnectFuture.cancel(false);
+			reconnectFuture = null;
+		}
+		statnode.setValue(new Value("connecting to device"));
+
+		String host = node.getAttribute(ATTR_HOST).getString();
+		int port = node.getAttribute(ATTR_PORT).getNumber().intValue();
+
+		getMasterAttributes();
```

Comment:
I'm not sure what that does, but if it has some side effect, maybe it shouldn't be named getSomething